lib/index.js

```diff
@@ -529,6 +530,22 @@ module.exports = {
 						var globalArrow = currentlyInGlobalScope(context.parserOptions,context.getScope());
 						var foundThis = thisFoundIn.has(node);
 
+						if (neverGlobalThis) {
+							if (foundThis && globalArrow) {
+								context.report({
+									node: node,
+									messageId: "neverGlobal",
+								});
+							}
+							else if (foundThis && nestedThis) {
```

Comment:
`nestedThis` will never be `true` in this context, since it's an exclusive mode alongside "never-global". Moreover, "never-global" sorta implies "nested" in this specific context, in that we're tracking even nested global arrows.

I'd simplify the logic here and restructure this whole `if..else if` structure like this:

```js
if (neverGlobalThis && foundThis) {
   if (globalArrow) {
      context.report(..);
   }
   else {
      let parentArrow = ..
      if (parentArrow) {
         // ..
      }
   }
}
```

Also, down on line 570, we need to exclude that block from also tracking parent arrow functions since we're handling that here:

```js
else if (!neverGlobalThis && foundThis && nestedThis) {
   // ..
```